opensbi-revyos: add recipe

OldManYellsAtCloud opened the pull request:
opensbi-revyos is an opensbi fork, specific for beaglev-ahead MACHINE.

<!--
Please make sure you've read and understood our contributing guidelines.

For additional information on the contribution guidelines:
https://wiki.yoctoproject.org/wiki/Contribution_Guidelines#General_Information

If this PR fixes an issue, make sure your description includes "fixes #xxxx".

If this PR connects to an issue, make sure your description includes "connected to #xxxx".

Please provide the following information:
-->

**- <recipename>: Short log / Statement of what needed to be changed.**
  
**-(Optional pointers to external resources, such as defect tracking)**
  
**-The intent of your change.**
  
**-(Optional, if it's not clear from above) how your change resolves the
issues in the first part.**
  
**-Tag line(s) at the end.**

**-Signed-off-by: Random J Developer <random@developer.example.org>**

OldManYellsAtCloud commented:
@kraj @RP38 @kanakshilledar FYI - if someone could test this just to see if it works, it would be helpful.

kanakshilledar reviewed (approved):
I tried building this, and is working on my machine.
```
$ bitbake core-image-minimalLoading cache: 100% |###########################################################| Time: 0:00:01
Loaded 1946 entries from dependency cache.
NOTE: Resolving any missing task queue dependencies

Build Configuration:
BB_VERSION           = "2.12.0"
BUILD_SYS            = "x86_64-linux"
NATIVELSBSTRING      = "universal"
TARGET_SYS           = "riscv64-poky-linux"
MACHINE              = "beaglev-ahead"
DISTRO               = "poky"
DISTRO_VERSION       = "5.2"
TUNE_FEATURES        = "riscv64"
meta
meta-poky
meta-yocto-bsp       = "master:bb30b14ff14f55e54997603b0d1ae1e5ff48f1c7"
meta-riscv           = "master:8c0feac879842c3b2ff436eee2a9126c419a0129"

Sstate summary: Wanted 0 Local 0 Mirrors 0 Missed 0 Current 1764 (0% match, 100% complete)00:00
Initialising tasks: 100% |######################################################| Time: 0:00:02
NOTE: Executing Tasks
NOTE: Tasks Summary: Attempted 3683 tasks of which 3683 didn't need to be rerun and all succeeded.
```
![image](https://github.com/user-attachments/assets/6e62630a-d62e-494e-b4c4-a0cb51313cca)

Tested-By: Kanak Shilledar <kanakshilledar111@protonmail.com>

kraj reviewed (approved):
lgtm

OldManYellsAtCloud commented:
Thanks for the screenshot, happy to see that it is working (and a bit surprised that it does out of the box). v0.9 is even older than I expected, but it explains the problems with modern recipes.

In the last push I just removed the `opensbi-revyos-payloads.inc` file, as I just copied it from oe-core. Instead I changed the `require` line to use the original from oe-core. There should be no functional change.

Though I mentioned on the mailing list that I don't intend to have this merged, actually I don't have a lot against it, assuming that everyone is happy with this. Just that I don't have this board, and while I can fix Yocto-related things, I can't fix board-specific issues in this recipe.

kanakshilledar commented:
@OldManYellsAtCloud I did a clean pull of poky, and I am getting this error when building the image
```
ERROR: ParseError at /home/kanak/yocto/sources/poky/meta/classes-recipe/kernel.bbclass:183: Could not inherit file classes/kernel-fitimage.bbclass                             | ETA:  --:--:--
ERROR: Parsing halted due to errors, see error messages above
```

If I comment that line then the issue goes away! It is not triggered when I am doing a simple x86 build in a different build directory.

OldManYellsAtCloud commented:
I think it is not related to this, rather it willl be this commit from a few hours ago: https://git.yoctoproject.org/poky/commit/meta/classes-recipe/kernel.bbclass?id=18861ef0eb23d4f8823955a6bff5d3fe89e191b6

Look at this ominous shortlog: `kernel.bbclass: remove support for type fitImage` 

Bleeding edge oe-core is like that. Use a stable branch? Or older commit?
